docs/proposals/20220825-azuremanaged-cluster-exp-graduation.md

```diff
@@ -0,0 +1,73 @@
+
+Once the proposal is accepted and merged into the Cluster API project as an endorsed set of provider recommendations, capz can audit the existing "`AzureManagedCluster`" experimental implementation for areas of disagreement with said recommendations. For each discovery of disagreement, there should be a defensible reason to matriculate a divergent capz Managed Kubernetes implementation into a post-experimental definition of "`AzureManagedCluster`", and ideally some form of sign-off from other provider maintainers. Where community consensus cannot be reached, capz should strongly consider evolving the experimental "`AzureManagedCluster`" implementation to meet the Cluster API Managed Kubernetes recommendation as a pre-requisite to graduating from experimental.
+
+### 3. Full AKS Feature Support
+
+"`AzureManagedCluster`" should be able to accommodate the entire feature set of AKS. Rather than require the concrete list of features of AKS to be fully implemented as a prerequisite to graduation, we should instead audit the AKS feature matrix against the "`AzureManagedCluster`" architectural surface area and ensure that capz is well prepared to continually integrate existing and new AKS features into "`AzureManagedCluster`" along a non-breaking path forward.
+
+### 4. Affinity between capz / Cluster API resource lifecycle enforcement and AKS lifecycle enforcement.
+
+Not unrelated to the above audit of AKS features, we should also overlay the Cluster API lifecycle combinatorics (simply speaking: Create, Read, Update, Delete) against the set of AKS cluster primitives to ensure that sane interfaces exist in capz in order to effectively enforce (for example, add an AKS node pool), or passively delegate authority (for example, defer to the built-in AKS autoscaler when enabled to enforce `MachinePool` replica count), if appropriate.
+
+### 5. Azure API Request Optimizations
+
+Prior to graduation from experimental, we should ensure that the sufficient set of configurable interfaces exist to allow "`AzureManagedCluster`" users to effectively tune capz so that no unnecessary Azure API requests are introduced into their AKS environment. We should ship sane, optimized defaults, but allow for flexible overrides to anticipate a wide variety of operational use cases.
+
+### 6. E2E Tests
+
+For every supported AKS feature, and for every supported lifecycle mutation of said feature, capz should have thorough, regular E2E test coverage.
+
+## Open Questions
+
+### Dependency upon (currently experimental) MachinePool spec
+
+At present we re-use the Cluster API `MachinePool` specification (as `AzureManagedMachinePool`) to implement AKS node pools running on Azure VMSS. A consideration here is that `MachinePool` is considered experimental, and behind a feature flag, by Cluster API. Do we want to add the graduation of `MachinePool` out of experimental as a prerequisite for graduating "`AzureManagedCluster`" out of experimental?
```

Comment:
Is there a proposal/kep that can be linked to for this? I think that'll make it easier for people to track the cross-project status.

Reply:
This is a good question. @mboersma @CecileRobertMichon is this the best current PR to reference for that effort?

https://github.com/kubernetes-sigs/cluster-api/pull/6089

Reply:
added link